src/core_plugins/kibana/public/management/sections/settings/components/field/field.js

```diff
@@ -328,7 +328,7 @@ export class Field extends PureComponent {
             label={!!unsavedValue ? 'On' : 'Off'}
             checked={!!unsavedValue}
             onChange={this.onFieldChange}
-            disabled={loading}
+            disabled={loading || setting.isControlledByServer}
```

Comment:
Maybe destructure `isControlledByServer` along with the other `setting` properties on line 322?